### DISCUSSION TITLE
Sonoff Basic R2 upgraded to 13.1 and will not connect to WiFi

### DISCUSSION THREAD

**Discussion by apastuszak** [arendst/Tasmota#19462] [Q&A]
I have a Sonoff Basic R2 that was on 12.2. I upgraded it to 13.1 using a minimal flash followed by sonoff.bin.gz.  After the bin flash, the Sonoff would not join the WiFi network it used to be on. It created its own network with the name of the device, which I could not connect to.

I reset the unit by power cycling 10 times and now it comes up with its own network named tasmota- followed by random letters and numbers.

If I connect to that network, I never get an IP address via DHCP. If I set the IP address manually to 192.168.4.100, I can then use my browser to navigate to 192.168.4.1, and the Wi-Fi configuration page loads. I enter my SSID and my password. The device reboots, but it never connects to my Wi-Fi.  It just sits there flashing it's little green light with no tasmota- W-Fi network. If I unplug it and plug it back in, it creates it's own tasmota- network.

My network name is SonoffNET and my password is 16 characters of just numbers, lowercase letters and uppercase letters.

I have another Sonoff Basic in the house that's on 12.1 that disconnected off the Wi-Fi and is not giving me similar issues.

All the other Sonoffs are working just fine.

I have rebooted the router and they still will not connect.

**Comment by phete67** [comment]
Today I discovered the same thing. Using the serial console, I can see the device try to continually connect to the wifi but fails. However, if I use 13.1 lite, it works fine.

**Comment by Jason2866** [comment]
Search in discussions and issues.

**Reply by apastuszak** [reply]
A search is not yielding anything useful, other than a discussion about Asus routers and temperature.

**Reply by sfromis** [reply]
A key takeaway from search should be that the Sonoff Basic R2 is a crappy device, barely able to power itself, and may be overloaded by full tasmota.

**Reply by apastuszak** [reply]
Well, it was working just fine on 12.2 and every version before that for probably the last 3-4 years for me. I know it's on me, because I flashed it.

I ordered 2 Shelly Plus 1s this morning and will try to use those and see if I like them better. But they are significantly more expensive.

I think I also want to check out the Sonoff 4CHAN Pro and see how that goes.

But all these devices use some variant of the ESP chip. I don't see why the ESP8266 on a Sonoff Basic would have WiFi issues with Tasmota, while the ESP8266 in some other device would not.

I like that the Shelly's are UL listed. Just need to see if I can find a cheaper US supplier than Amazon for them.

**Reply by Jason2866** [reply]
Two possibilitys you have
- do really search for Sonoff Basic Issues
- trust us that specific devices have Hardware problems

**Reply by apastuszak** [reply]
I was hoping my search would yield a solution. Obviously it will not. I'll try to downgrade to 12.2 over the serial interface and see what happens.

**Reply by sfromis** [reply]
The issue is not up the ESP8266, but the rest of the bad hardware in the Sonoff Basic R2, not able to supply enough power to the ESP8266. And with Tasmota having grown more feature-rich, it may also do more work, making the ESP8266 need a bit more power.

**Reply by apastuszak** [reply]
SO, recommend me a simple switch .

**Reply by Jason2866** [reply]
Pick your poison https://templates.blakadder.com/
Every device is better than a Sonoff Basic R2

**Comment by apastuszak** [comment]
I broke down today and serially flashed 13.1 onto this device and it came back to life. But while doing so, I broke the button on it. Then the metal pad over GPIO 0 (or is it 1) that you need to push down on to program it fell off.

So, serial flashing is going to be a bit of a PITA now.

I set the device back up in Home Assistant, configured two home automations and tested them. And after I tried the first automation, the device became unresponsive. I power cycled the device, and I am not even getting the flashing green light.

So, it's dead.

Time to wire one of the Shellys up and see how it goes.

**Comment by dirk2947** [comment]
Hey there, just wanted to confirm your findings. 100% accurate. I wanted to tear my hair out for hours. I thought something was wrong with my wifi adapter / network settings because I simply couldn't get them to save my network settings / connect to my network.
Now I'm back at 9.1 which works flawlessly. I should just buy new switches...

**Comment by pezoli** [comment]
If you dont need sensors, just the switch function via MQTT, you can upgrade to latest release, just don't upgrade to full firmware (tasmota.bin), instead use tasmota-lite.bin. Maybe this uses less power, my Sonoff R1 switches are working fine this way.
